default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml

```diff
@@ -49,3 +49,36 @@
     url: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-6.0-Release-Notes#web-applications
   - title: 'Specific Spring Framwork change'
     url: https://github.com/spring-projects/spring-framework/issues/28552
+
+- ruleID: spring-framework-5.x-to-6.0-web-applications-00020
+  category: mandatory
+  effort: 5
+  labels:
+  - konveyor.io/source=spring5
+  - konveyor.io/target=spring6+
+  when:
+    and:
+    - java.referenced:
+        pattern: org.springframework.web.client.RestTemplate
```

Comment:
We should only look for the usage of [the `HttpComponentsClientHttpRequestFactory` class](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.html), which is the one that requires Apache HttpClient 5. In our case, it is likely to be found as a parameter for [the `RestTemplate(ClientHttpRequestFactory requestFactory)` constructor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html#%3Cinit%3E(org.springframework.http.client.ClientHttpRequestFactory)). Only looking for the RestTemplate class could lead to false positives.